File: nestjs-backend/src/rooms/dto/create-room.dto.ts
import { IsNotEmpty, IsString, Length, IsOptional, IsBoolean } from 'class-validator';

/**
 * Data Transfer Object for creating chat rooms
 * 
 * Validates room creation requests from REST API clients
 * and ensures proper room configuration.
 */
export class CreateRoomDto {
  @IsNotEmpty({ message: 'Room name is required' })
  @IsString({ message: 'Room name must be a string' })
  @Length(3, 100, { message: 'Room name must be between 3 and 100 characters' })
  name: string;

  @IsOptional()
  @IsString({ message: 'Room description must be a string' })
  @Length(0, 500, { message: 'Room description must not exceed 500 characters' })
  description?: string;

  @IsOptional()
  @IsBoolean({ message: 'isPrivate must be a boolean' })
  isPrivate?: boolean = false;
}

/**
 * DTO for updating room information
 */
export class UpdateRoomDto {
  @IsOptional()
  @IsString({ message: 'Room name must be a string' })
  @Length(3, 100, { message: 'Room name must be between 3 and 100 characters' })
  name?: string;

  @IsOptional()
  @IsString({ message: 'Room description must be a string' })
  @Length(0, 500, { message: 'Room description must not exceed 500 characters' })
  description?: string;

  @IsOptional()
  @IsBoolean({ message: 'isPrivate must be a boolean' })
  isPrivate?: boolean;
}

File: spring-boot-backend/src/main/java/com/chatapp/dto/CreateRoomDto.java
package com.chatapp.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;

/**
 * Data Transfer Object for creating chat rooms
 * 
 * Used for validating room creation requests from REST API clients
 * and ensuring proper room configuration.
 */
public class CreateRoomDto {
    
    @NotBlank(message = "Room name is required")
    @Size(min = 3, max = 100, message = "Room name must be between 3 and 100 characters")
    private String name;
    
    @Size(max = 500, message = "Room description must not exceed 500 characters")
    private String description;
    
    private boolean isPrivate = false;

    /**
     * Default constructor
     */
    public CreateRoomDto() {
    }

    /**
     * Constructor with parameters
     * 
     * @param name Display name of the room
     * @param description Optional room description
     * @param isPrivate Whether the room is private
     */
    public CreateRoomDto(String name, String description, boolean isPrivate) {
        this.name = name;
        this.description = description;
        this.isPrivate = isPrivate;
    }

    // Getters and Setters
    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public boolean isPrivate() {
        return isPrivate;
    }

    public void setPrivate(boolean isPrivate) {
        this.isPrivate = isPrivate;
    }

    @Override
    public String toString() {
        return "CreateRoomDto{" +
                "name='" + name + '\'' +
                ", description='" + description + '\'' +
                ", isPrivate=" + isPrivate +
                '}';
    }
}